dll/win32/browseui/shellfind/CSearchBar.cpp

```diff
@@ -131,6 +135,29 @@ LRESULT CSearchBar::OnSearchButtonClicked(WORD wNotifyCode, WORD wID, HWND hWndC
         return 0;
     }
 
+    // See if we have a szFileName, our first character is not an asterisk,
+    // and we have room in MAX_PATH to add an asterisk at the beginning of szFileName then add it
+    StringCchLength(pSearchStart->szFileName, MAX_PATH, &len);
```

Comment:
MSDN says for the last parameter:
> This value is valid only if pcch is not NULL and the function succeeds.

I know this isn't supposed to fail but unlike the `StringCchCatW` below, the failure mode here sucks (uninit variable use).
Maybe `if (FAILED(StringCchLengthW(...)) return 0;`? (or otherwise properly assign it to an `HRESULT` variable and assert on it?)

Reply:
Done. Thanks.